eventbus.go

```diff
@@ -50,9 +60,19 @@ func (e EventBus) Publish(eventName string, data Data) {
 		return
 	}
 
-	for _, subscriber := range subscribers {
-		subscriber <- data
+	go func(data Data, subscribers []DataChannel) {
+		for _, ch := range subscribers {
+			ch <- data
+		}
+	}(data, subscribers)
+}
+
+func swapSubscribers(i, j int, subscribers []DataChannel) {
+	if len(subscribers)-1 < i || len(subscribers)-1 < j {
```

Comment:
```suggestion
	if i >=  len(subscribers)  ||  j >= len(subscribers) {
```
they are the same but I think the suggestion is more clear